diff --git a/meson.build b/meson.build
index b854155..d124d77 100644
--- a/meson.build
+++ b/meson.build
@@ -11,8 +11,8 @@ gnome = import('gnome')
 
 base_id = 'io.github.seadve.Kooha'
 
-dependency('glib-2.0', version: '>= 2.80')
-dependency('gio-2.0', version: '>= 2.80')
+dependency('glib-2.0', version: '>= 2.79')
+dependency('gio-2.0', version: '>= 2.79')
 dependency('gtk4', version: '>= 4.14')
 dependency('libadwaita-1', version: '>= 1.5')
 dependency('gstreamer-1.0', version: '>= 1.22')
