diff --git a/src/decoder/formats/svg.rs b/src/decoder/formats/svg.rs
index cdc3871..31a659d 100644
--- a/src/decoder/formats/svg.rs
+++ b/src/decoder/formats/svg.rs
@@ -71,7 +71,7 @@ impl Svg {
         updater.clone().spawn_error_handled(async move {
             log::trace!("Setting up SVG loader");
             let mut loader = glycin::Loader::new(file);
-            loader.sandbox_selector(glycin::SandboxSelector::FlatpakSpawn);
+            loader.sandbox_selector(glycin::SandboxSelector::Auto);
 
             loader.cancellable(cancellable.clone());
 
